### ISSUE TITLE
setCredentials should be able to support connection w/o TLS cert

### ISSUE THREAD

**Issue by bucko13** [Tierion/lnrpc-node-client#3]
This should enable support for connecting with a BTCPay Server node. Since they just use port `443` and https for connections, there is no TLS cert for connecting to the node. Just the macaroon and the socket. 

Being able to connect to BTCPay Server nodes adds a lot more flexibility of what node people can use for chianpoint, especially since these are nodes that can be run relatively cheaply and can be setup with very little coding experience. 

I imagine it would be a pretty simple change to this line:
https://github.com/Tierion/lnrpc-node-client/blob/master/src/index.js#L43

If these seems reasonable, I can put together a PR for this change.

**Comment by jacohend** [comment]
Seems like just a matter of detecting the case when the socket is a https url and the provided tlscert variable is null.

**Comment by bucko13** [comment]
Yep, that's a possibility. Also just making TLS optional could do it too. I guess it depends on what use cases you want to allow vs. ones you want to block. It's possible to have a lightning node with no TLS and not behind https for example. From what I could gather from clients like Zap, when connecting to a BTCPay Server, it just ignores the TLS when setting the grpc client. This is what I added to boltwall too (in order to support btcpay)

**Comment by jacohend** [comment]
I think we might want to avoid the specific case of no TLS _and_ no https, but I don't see a problem with supporting any other case.  I think in general we should encourage encrypted connections if the connection is nonlocal.
